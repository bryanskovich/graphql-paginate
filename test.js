const graphqlPagination = require("./index");

describe("pagination helper", () => {
  const models = [
    {
      id: 1,
      name: "user 1"
    },
    {
      id: 2,
      name: "user 2"
    },
    {
      id: 3,
      name: "user 3"
    },
    {
      id: 4,
      name: "user 4"
    },
    {
      id: 5,
      name: "user 5"
    }
  ];
  const mockModel = {
    findAll: jest.fn(() => models),
    count: jest.fn(() => models.length)
  };

  it("call skip , first and limit args ", async () => {
    const args = {
      skip: 2,
      first: 2,
      LIMIT: 2
    };
    const pagination = await graphqlPagination(mockModel, args, "id");

    expect(pagination).toMatchSnapshot();
  });
});
