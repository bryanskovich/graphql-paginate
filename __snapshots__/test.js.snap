// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination helper call skip , first and limit args  1`] = `
Object {
  "edges": Array [
    Object {
      "cursor": 1,
      "node": Object {
        "id": 1,
        "name": "user 1",
      },
    },
    Object {
      "cursor": 2,
      "node": Object {
        "id": 2,
        "name": "user 2",
      },
    },
  ],
  "pageInfo": Object {
    "endCursor": 2,
    "hasNextPage": true,
    "hasPreviousPage": true,
    "startCursor": 1,
  },
  "totalCount": 5,
}
`;
